File: src/main/java/com/moovie/backend/domain/usecase/movies/GetMovieListDetail.java
package com.moovie.backend.domain.usecase.movies;

import com.moovie.backend.domain.model.MovieListDetail;
import com.moovie.backend.domain.repository.MoviesRepository;
import org.springframework.stereotype.Service;

@Service
public class GetMovieListDetail {

    private final MoviesRepository moviesRepository;

    public GetMovieListDetail(MoviesRepository moviesRepository) {
        this.moviesRepository = moviesRepository;
    }

    public MovieListDetail invoke(int listId, int page) {
        return moviesRepository.getMovieListDetail(listId, page);
    }
}

File: src/main/kotlin/org/example/domain/usecase/movies/GetMovieListDetail.kt
package org.example.domain.usecase.movies

import kotlinx.coroutines.runBlocking
import org.example.domain.model.MovieListDetail
import org.example.domain.repository.MoviesRepository

class GetMovieListDetail(private val repository: MoviesRepository) {
    operator fun invoke(listId: Int, page: Int): MovieListDetail = runBlocking {
        repository.getMovieListDetail(listId, page)
    }
}
